akd_core/src/utils.rs

```diff
@@ -164,32 +276,115 @@ macro_rules! test_config_sync {
 mod tests {
     use super::*;
     use alloc::vec;
+    use rand::{rngs::OsRng, Rng};
 
     #[test]
     fn test_get_marker_versions() {
         assert_eq!(
-            (vec![], vec![6, 7, 8, 16, 32]),
-            get_marker_versions(1, 5, 33)
+            (vec![16, 64], vec![66, 68, 72, 80, 96, 128]),
+            get_marker_versions(65, 65, 128)
         );
-
         assert_eq!(
-            (vec![], vec![6, 7, 8, 16, 32]),
-            get_marker_versions(2, 5, 33)
+            (vec![16, 64, 80, 84], vec![86, 88, 96, 128, 256, 65536]),
+            get_marker_versions(85, 85, 65537)
         );
+        assert_eq!((vec![], vec![6, 8, 16]), get_marker_versions(1, 5, 33));
 
-        assert_eq!(
-            (vec![2], vec![6, 7, 8, 16, 32]),
-            get_marker_versions(3, 5, 33)
-        );
+        assert_eq!((vec![], vec![6, 8, 16]), get_marker_versions(2, 5, 33));
+
+        assert_eq!((vec![2], vec![6, 8, 16]), get_marker_versions(3, 5, 33));
+
+        assert_eq!((vec![4], vec![13, 14, 16]), get_marker_versions(6, 12, 128));
 
         assert_eq!(
-            (vec![4], vec![13, 14, 15, 16, 32, 64, 128]),
-            get_marker_versions(6, 12, 128)
+            (vec![4], vec![13, 14, 16, 256]),
+            get_marker_versions(6, 12, 256)
         );
 
         assert_eq!(
-            (vec![4], vec![13, 14, 15, 16, 32, 64]),
-            get_marker_versions(6, 12, 127)
+            (vec![16, 128], vec![131, 132, 136, 144, 160, 192, 256]),
+            get_marker_versions(130, 130, 256)
         );
     }
+
+    #[derive(Clone)]
+    enum RangeType {
+        Small,
+        Medium,
+        Large,
+    }
+
+    fn gen_versions(
+        rng: &mut OsRng,
+        start_type: &RangeType,
+        end_type: &RangeType,
+        epoch_type: &RangeType,
+    ) -> (u64, u64, u64) {
+        let small_jump = 10;
+        let medium_jump = 1000;
```

Comment:
Think it makes sense to tie the numeric values to the enum entries as part of the type, or as a conversion into an i32 or something? Might make it a bit more explicit rather than putting those details in this fn. Feel free to ignore if you prefer this!

Reply:
There was some annoying math / edge cases to cover with these ranges that I ended up hardcoding with these as integers, and since it's test code, I figured not to bother too much in terms of over-engineering it. I think this is ok for now.